src/main/java/com/spotify/docker/client/messages/HostConfig.java

```diff
@@ -1036,7 +1038,12 @@ public String toString() {
       } else if (from == null || from.equals("")) {
         return to;
       } else if (readOnly == null || !readOnly) {
-        return from + ":" + to;
+          if (noCopy == null || !noCopy) {
```

Comment:
I checked out [the volume options](https://docs.docker.com/engine/reference/run/#/volume-shared-filesystems) after reading docker/docker#27046. It isn't worded very clearly, but it does say you can set `ro` and `nocopy` together, along with any of the other options.
> The **comma-delimited** `options` are are `[rw|ro]`, `[z|Z]`, `[[r]shared|[r]slave|[r]private]`, and `[nocopy]`.

Emphasis mine.


```bash
$ docker volume create --name foo
foo
$ docker run -v foo:/tmp:ro:nocopy busybox
docker: Error response from daemon: Invalid bind mount spec "foo:/tmp:ro:nocopy": Invalid volume specification: 'foo:/tmp:ro:nocopy'.
See 'docker run --help'.
$ docker run -v foo:/tmp:ro,nocopy busybox
[no error]
```